src/login/logind-varlink.c

```diff
@@ -200,6 +200,15 @@ static int vl_method_create_session(sd_varlink *link, sd_json_variant *parameter
                         return sd_varlink_error(link, "io.systemd.Login.NoSuchSeat", /* parameters= */ NULL);
         }
 
+        _cleanup_free_ char *resolved = NULL;
+        if (tty_is_console(p.tty)) {
+                r = resolve_dev_console(&resolved);
+                if (r < 0)
+                        log_debug_errno(r, "Failed to resolve /dev/console, ignoring: %m");
+                else
+                        p.tty = resolved;
+        }
```

Comment:
I'd move this into the p.tty if branch, and apply the fallback logic if it can't be resolved, rather than testing is_console() again below

Reply:
(nevermind, we'd end up with redundant tty_is_vc call that way)